pytests/test_bjoern.py

```diff
@@ -0,0 +1,464 @@
+# -*- coding: utf-8 -*-
+
+import contextlib
+import json
+import math
+import os
+import unittest
+import signal
+import socket
+import sys
+import textwrap
+from wsgiref.validate import validator
+
+import pytest
+import requests
+import bjoern
+
+
+@pytest.mark.parametrize('header_x_auth', ('X-Auth_User', 'X_Auth_User', 'X_Auth-User'))
+def test_CVE_2015_0219(client, header_x_auth):
+    """
+    https://www.djangoproject.com/weblog/2015/jan/13/security/
+    https://snyk.io/vuln/SNYK-PYTHON-BJOERN-40507
+    """
+
+    def app(env, start_response):
+        assert header_x_auth not in env.keys()
+        assert 'X-Auth-User' not in env.keys()
+        start_response('200 ok', [])
+        return b''
+
+    client.start(app)
+    client.get(headers={header_x_auth: 'admin'})
+
+
+def test_listen(client):
+    """tests/listen.py"""
+
+    def app(environ, start_response):
+        start_response('200 OK', [])
+        yield b'Hello world'
+        yield b''
+
+    client.start(app)
+    assert client.get().content == b'Hello world'
+
+
+@pytest.mark.parametrize(
+    'content,status_code',
+    (
+        ('200 ok', 200),
+        ('201 created', 201),
+        ('202 accepted', 202),
+        ('204 no content', 204),
+    ),
+)
+def test_status_codes(client, content, status_code):
+    """tests/204.py"""
+
+    def app(e, s):
+        s(content, [])
+        return b''
+
+    client.start(app)
+    resp = client.get()
+    assert resp.status_code == status_code
+
+
+def test_empty_content(client):
+    """tests/empty.py"""
+
+    def app(e, s):
+        s('200 ok', [])
+        return b''
+
+    client.start(app)
+    response = client.get()
+    assert response.content == b''
+
+
+@pytest.mark.parametrize('method', ('DELETE', 'GET', 'OPTIONS', 'POST', 'PATCH', 'PUT'))
+def test_env_request_method(client, method):
+    """tests/env.py"""
+
+    def app(env, start_response):
+        start_response('200 yo', [])
+        assert env['REQUEST_METHOD'] == method
+        return []
+
+    client.start(app)
+    m = getattr(client, method.lower())
+    m()
+
+
+def wrap(text, width=20, placeholder='...'):
+    wrapped = textwrap.wrap(text, width=width)[0]
+    if len(wrapped) < len(text):
+        wrapped += placeholder
+    return wrapped
+
+
+@pytest.mark.parametrize(
+    'header_key,header_value',
```

Comment:
This decorator is very difficult to read, maybe you can reformat

Reply:
This comes from the `black` autoformatter which runs on every commit on my machine. I will try configure `black` to use a bigger line length argument, the formatting is indeed questionable...